litrpc/netcmds.go

```diff
@@ -64,6 +65,29 @@ func (r *LitRPC) Connect(args ConnectArgs, reply *StatusReply) error {
 	return nil
 }
 
+// ------------------------- name a connection
+type AssignNicknameArgs struct {
+	Peer     uint32
+	Nickname string
+}
+
+func (r *LitRPC) AssignNickname(args AssignNicknameArgs, reply *StatusReply) error {
+	err := r.Node.SaveNicknameForPeerIdx(args.Nickname, args.Peer)
+	if err != nil {
+		return err
+	}
+
+	peer, ok := r.Node.RemoteCons[args.Peer]
+	if !ok {
+		return fmt.Errorf("peer %d doesn't exist", args.Peer)
```

Comment:
sorry, looking at this further, this isn't actually an error.  If the first call to `SaveNicknameForPeerIdx() ` fails, that's an error.  But of the second part, modifying `RemoteCons[args.Peer]` fails, that just means you assigned a nickname for a peer you're not currently connected to, which is OK.

So maybe it's better to move the `reply.Status` above this, and then just `if !ok {return nil}`